packages/ts-docs-gen/src/generator-helpers.ts

```diff
@@ -131,34 +129,56 @@ export namespace GeneratorHelpers {
 
     export function GetApiItemReferences(
         extractedData: ExtractDto,
-        itemsReference: Contracts.ApiItemReferenceTuple
+        itemsReference: Contracts.ApiItemReference[]
     ): ApiItemReference[] {
         let list: ApiItemReference[] = [];
 
-        for (const [alias, references] of itemsReference) {
-            for (const referenceId of references) {
+        for (const item of itemsReference) {
+            for (const referenceId of item.Ids) {
                 // Check if item is ExportSpecifier or ExportDeclaration.
                 const apiItem = extractedData.Registry[referenceId];
 
                 switch (apiItem.ApiKind) {
                     case Contracts.ApiItemKinds.Export: {
-                        const referenceTuples = GetApiItemReferences(extractedData, apiItem.Members);
-                        list = list.concat(referenceTuples);
+                        if (apiItem.SourceFileId != null) {
+                            const sourceFileReference = { Alias: apiItem.Name, Ids: [apiItem.SourceFileId] };
+                            const referencesList = GetApiItemReferences(extractedData, [sourceFileReference]);
+                            list = list.concat(referencesList);
+                        }
                         break;
                     }
+                    case Contracts.ApiItemKinds.ImportSpecifier:
                     case Contracts.ApiItemKinds.ExportSpecifier: {
                         if (apiItem.ApiItems == null) {
-                            console.warn(`ApiItems are missing in "${apiItem.Name}"?`);
+                            LogWithApiItemPosition(
+                                LogLevel.Warning,
+                                apiItem,
+                                `ApiItems are missing in "${apiItem.Name}"?`
+                            );
                             break;
                         }
-                        const referenceTuples = GetApiItemReferences(extractedData, [[apiItem.Name, apiItem.ApiItems]]);
-                        list = list.concat(referenceTuples);
+
+                        const referencesList = GetApiItemReferences(extractedData, [{ Alias: apiItem.Name, Ids: apiItem.ApiItems }]);

```

Comment:
Move to separate `const` just like in [line 144](https://github.com/SimplrJS/ts-docs-gen/pull/9/files#diff-204caa7f4a15f6b638af4acdaf73b5fdR144).